Lib/test/test_concurrent_futures.py

```diff
@@ -178,6 +178,20 @@ def test_thread_names_default(self):
             self.assertRegex(t.name, r'^.*ThreadPoolExecutor.*_[0-4]$')
             t.join()
 
+    def test_default_max_queue_size(self):
+        executor = futures.ThreadPoolExecutor()
+        self.assertEqual(executor._work_queue.maxsize, 0)
+
+    def test_custom_max_queue_size(self):
+        # test custom valid and invalid values
+        for i in range(-10, 10):
+            executor = futures.ThreadPoolExecutor(max_queue_size=i)
+            self.assertEqual(executor._work_queue.maxsize, i)
+        # provided value is forwarded without checking it
```

Comment:
I am surprised by this. @rhettinger, is there any reason `queue.Queue` doesn't check for the size type?
```
>>> q = Queue(maxsize="foo")
>>> q.put(1)
Traceback (most recent call last):
  File "<ipython-input-4-f7b62bda2b8c>", line 1, in <module>
    q.put(1)
  File "/home/antoine/miniconda3/envs/dask35/lib/python3.5/queue.py", line 127, in put
    if self.maxsize > 0:
TypeError: unorderable types: str() > int()
```